syntax = "proto2";

package benchmarks.proto;

option cc_enable_arenas = true;

message TestAllTypes {
  optional bool bool_value = 1;
  optional int32 int32_value = 2;
  optional int64 int64_value = 3;
  optional string string_value = 4;
  optional bytes bytes_value = 5;

  repeated bool repeated_bool_value = 101;
  repeated int32 repeated_int32_value = 102;
  repeated int64 repeated_int64_value = 103;
  repeated string repeated_string_value = 104;
  repeated bytes repeated_bytes_value = 105;

  repeated TestAllTypes children = 201;
}
